### PR TITLE
Flesh out the 1st best practice: "Components Cannot Apply External Spacing"

### PR DESCRIPTION
**Pull request by chris-pearce**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,99 @@
...
+# UI Component Best Practices <!-- omit in toc -->
+
+A bunch of best practices to help achieve User Interface (UI) components, or
+presentational components, that are:
+
+- accessible
+- consumable
+- portable
+- reusable
+- and robust.
```
Comment (DaveOrDead) [inline README.md]: As this is a bullet list, could probably just say `robust` without `and` and the `.`

File: README.md [hunk 2]
```diff
@@ -1 +1,99 @@
...
+
+## Table of Contents <!-- omit in toc -->
+
+- [Caveats](#caveats)
+- [The tl;dr Version](#the-tldr-version)
+- [Components Cannot Apply External Spacing](#components-cannot-apply-external-spacing)
```
Comment (DaveOrDead) [inline README.md]: `should not` might be a better way of phrasing this, as there is an exceptions section

File: README.md [hunk 3]
```diff
@@ -1 +1,99 @@
...
+  - [The Rule](#the-rule)
+  - [Why?](#why)
+    - [Why `margin` Specifically?](#why-margin-specifically)
+  - [The Exception(s)](#the-exceptions)
+  - [More Reading](#more-reading)
+
+## Caveats
+
+1. These best practices target components that are part of an organisation's component library or broader design system where the design is known.
+2. All code snippets are from the [React library](https://reactjs.org/).
```
Comment (DaveOrDead) [inline README.md]: Do you mean you are using code snippets from their site, or that they are using JSX?

File: README.md [hunk 4]
```diff
@@ -1 +1,99 @@
...
+   However, this checklist is library/framework-agnostic. Meaning it can apply
```
Comment (DaveOrDead) [inline README.md]: This feels like one sentence to me

File: README.md [hunk 5]
```diff
@@ -1 +1,99 @@
...
+```jsx
+<Search />
+<SearchResults />
+<Pagination />
+<Downloads />
+```
+
+There needs to be `40px` of space between each component. We can easily achieve
+this by adding `margin-bottom: 40px;` to each of the component's root element.
+However, that would be short sighted as the
```
Comment (DaveOrDead) [inline README.md]: Seems like your lines are breaking prematurely in a few spots

File: README.md (hunk 1 above)
Comment (chris-pearce) [reply README.md]: I was using the list in sentence form. It was missing the commas after the list items bar the last one. And I formatted "and" a little differently:

```
- accessible,
- consumable,
- portable,
- reusable, and
- robust.
```

File: README.md (hunk 3 above)
Comment (chris-pearce) [reply README.md]: I updated it to make it clearer.

File: README.md (hunk 5 above)
Comment (chris-pearce) [reply README.md]: Added Prettier.

**Comment by chris-pearce** [comment]
@DaveOrDead thanks for the feedback, I've applied all your feedback.

**Pull request merged**